File: check_map.js
function mapa_cerrado(mapa) {
    let filas = mapa.length;
    let columnas = mapa[0].length;
    
    // Verificar bordes del mapa
    for (let i = 0; i < filas; i++) {
      if (mapa[i][0] === 32) {
        if (!hay_camino(mapa, i, 0)) {
          return false;
        }
      }
      if (mapa[i][columnas-1] === 32) {
        if (!hay_camino(mapa, i, columnas-1)) {
          return false;
        }
      }
    }
    
    for (let j = 0; j < columnas; j++) {
      if (mapa[0][j] === 32) {
        if (!hay_camino(mapa, 0, j)) {
          return false;
        }
      }
      if (mapa[filas-1][j] === 32) {
        if (!hay_camino(mapa, filas-1, j)) {
          return false;
        }
      }
    }
    
    // Mapa está cerrado
    return true;
  }
  
  function hay_camino(mapa, i, j) {
    let filas = mapa.length;
    let columnas = mapa[0].length;
    
    let visitado = new Array(filas).fill().map(() => new Array(columnas).fill(false));
    let queue = [[i, j]];
    
    while (queue.length > 0) {
      let [x, y] = queue.shift();
      if (x === 0 || x === filas-1 || y === 0 || y === columnas-1) {
        return false;  // Se llegó al borde del mapa
      }
      
      if (visitado[x][y]) {
        continue;
      }
      visitado[x][y] = true;
      
      if (mapa[x-1][y] === 32) {
        queue.push([x-1, y]);
      }
      if (mapa[x+1][y] === 32) {
        queue.push([x+1, y]);
      }
      if (mapa[x][y-1] === 32) {
        queue.push([x, y-1]);
      }
      if (mapa[x][y+1] === 32) {
        queue.push([x, y+1]);
      }
    }
    
    return true;  // No se llegó al borde del mapa, hay un camino hacia el exterior
  }
  
  // Matriz de enteros correspondiente al nuevo mapa
  let mapa_nuevo =
                [
                    [32, 32, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 32, 32, 32, 32],
                    [32, 49, 49, 49, 48, 48, 48, 48, 48, 48, 48, 49, 49, 49, 32, 32],
                    [32, 49, 49, 48, 48, 48, 48, 48, 48, 48, 48, 48, 49, 49, 49, 32],
                    [32, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 32, 32],
                    [32, 49, 49, 32, 49, 32, 49, 32, 49, 49, 49, 49, 49, 32, 32, 32]
                ]

if (mapa_cerrado(mapa_nuevo))
    console.log("El mapa está cerrado correctamente por 1");
else
    console.log("El mapa no está cerrado correctamente por 1");

File: check_map.py
def mapa_cerrado(mapa):
    filas = len(mapa)
    columnas = len(mapa[0])
    
    # Verificar bordes del mapa
    for i in range(filas):
        if mapa[i][0] == ' ':
            if not hay_camino(mapa, i, 0):
                return False
        if mapa[i][columnas-1] == ' ':
            if not hay_camino(mapa, i, columnas-1):
                return False
    
    for j in range(columnas):
        if mapa[0][j] == ' ':
            if not hay_camino(mapa, 0, j):
                return False
        if mapa[filas-1][j] == ' ':
            if not hay_camino(mapa, filas-1, j):
                return False
    
    # Mapa está cerrado
    return True

def hay_camino(mapa, i, j):
    filas = len(mapa)
    columnas = len(mapa[0])
    
    visitado = [[False for _ in range(columnas)] for _ in range(filas)]
    queue = [(i, j)]
    
    while queue:
        x, y = queue.pop(0)
        if x == 0 or x == filas-1 or y == 0 or y == columnas-1:
            return False  # Se llegó al borde del mapa
        
        if visitado[x][y]:
            continue
        visitado[x][y] = True
        
        if mapa[x-1][y] == ' ':
            queue.append((x-1, y))
        if mapa[x+1][y] == ' ':
            queue.append((x+1, y))
        if mapa[x][y-1] == ' ':
            queue.append((x, y-1))
        if mapa[x][y+1] == ' ':
            queue.append((x, y+1))
    
    return True  # No se llegó al borde del mapa, hay un camino hacia el exterior

# Matriz de enteros correspondiente al nuevo mapa
mapa_nuevo = [  [32, 32, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 32, 32, 32, 32],
                [32, 49, 49, 49, 48, 48, 48, 48, 48, 48, 48, 49, 49, 49, 32, 32],
                [32, 49, 49, 48, 48, 48, 48, 48, 48, 48, 48, 48, 49, 49, 49, 32],
                [32, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 49, 32, 32],
                [32, 49, 49, 32, 49, 32, 49, 32, 49, 32, 49, 49, 49, 32, 32, 32]
]

if mapa_cerrado(mapa_nuevo):
    print("El mapa está cerrado correctamente por 1")
else:
    print("El mapa no está cerrado correctamente por 1")